### PR TITLE
feat(frontend): upload a new avatar bb-341

### PR DESCRIPTION
**Pull request by MaxKorop**
### Summary
The user needs to upload a new avatar from the user profile screen. The form for the avatar is separated from the form for updating the user details.
Used real user avatar in all other places as well (example: header)

### Screenshots
image

### CODE REVIEW

File: apps/frontend/src/libs/modules/api/base-http-api.ts [hunk 1]
```diff
@@ -110,6 +110,10 @@ class BaseHTTPApi implements HTTPApi {
 
 		const headers = await this.getHeaders(contentType, hasAuth);
 
+		if (payload instanceof FormData) {
+			headers.delete(HTTPHeader.CONTENT_TYPE);
+		}
```
Comment (fshabanov) [inline apps/frontend/src/libs/modules/api/base-http-api.ts:115]: Why not add multipart/form-data?

File: apps/frontend/src/pages/profile/libs/components/update-avatar-form/update-avatar-form.tsx [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+	onSubmit,
+	user,
+}: Properties) => {
+	const fileInputReference = useRef<HTMLInputElement>(null);
+
+	const handleFormChange: React.FormEventHandler = useCallback(
+		(event: React.ChangeEvent<HTMLInputElement>): void => {
+			event.preventDefault();
+			const formData = new FormData();
+			formData.append("", event.target.files?.item(ZERO_INDEX) as File);
```
Comment (fshabanov) [inline apps/frontend/src/pages/profile/libs/components/update-avatar-form/update-avatar-form.tsx:25]:
```suggestion
			formData.append("file", event.target.files?.item(ZERO_INDEX) as File);
```
And move that name to a separate enum

File: packages/shared/src/libs/enums/content-type.enum.ts [hunk 3]
```diff
@@ -1,6 +1,7 @@
 const ContentType = {
 	JPEG: "image/jpeg",
 	JSON: "application/json",
+	MULTIPART: "multipart/form-data; boundary=abcdef",
```
Comment (fshabanov) [inline packages/shared/src/libs/enums/content-type.enum.ts:4]:
```suggestion
	MULTIPART_FORM_DATA: "multipart/form-data;",
```

**Review by fshabanov — CHANGES_REQUESTED** [review]

File: apps/frontend/src/libs/modules/api/base-http-api.ts (hunk 1 above)
Comment (fshabanov) [reply apps/frontend/src/libs/modules/api/base-http-api.ts:115]: If you remove it later, why do you add it in the first place?

Comment (MaxKorop) [reply apps/frontend/src/libs/modules/api/base-http-api.ts:115]: Do you mean why I added it to the ContentType but then deleted it before sending the request? Then I can remove this field in this enum, and pass ContentType.JSON to the request, then this header will be removed and automatically set multipart/form-data when requested

Comment (fshabanov) [reply apps/frontend/src/libs/modules/api/base-http-api.ts:115]: I can't understand, why do you want to pass application/json? In getHeaders add a check and don't add the header, if it is multipart

File: apps/frontend/src/libs/modules/api/base-http-api.ts [hunk 4]
```diff
@@ -66,8 +66,11 @@ class BaseHTTPApi implements HTTPApi {
 		hasAuth: boolean,
 	): Promise<Headers> {
 		const headers = new Headers();
+		const isMultipartFormData = contentType !== ContentType.MULTIPART_FORM_DATA;
```
Comment (fshabanov) [inline apps/frontend/src/libs/modules/api/base-http-api.ts:69]:
```suggestion
		const isMultipartFormData = contentType === ContentType.MULTIPART_FORM_DATA;
```

File: apps/frontend/src/libs/modules/api/base-http-api.ts [hunk 5]
```diff
@@ -66,8 +66,11 @@ class BaseHTTPApi implements HTTPApi {
...
 
-		headers.append(HTTPHeader.CONTENT_TYPE, contentType);
+		if (isMultipartFormData) {
```
Comment (fshabanov) [inline apps/frontend/src/libs/modules/api/base-http-api.ts:71]:
```suggestion
		if (!isMultipartFormData) {
```

File: apps/frontend/src/pages/profile/libs/components/update-avatar-form/update-avatar-form.tsx [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+	onSubmit,
+	user,
+}: Properties) => {
+	const fileInputReference = useRef<HTMLInputElement>(null);
+
+	const handleFormChange: React.FormEventHandler = useCallback(
+		(event: React.ChangeEvent<HTMLInputElement>): void => {
+			event.preventDefault();
+			const formData = new FormData();
+			formData.append("file", event.target.files?.item(ZERO_INDEX) as File);
```
Comment (fshabanov) [inline apps/frontend/src/pages/profile/libs/components/update-avatar-form/update-avatar-form.tsx:25]: What if no files were selected?

**Review by fshabanov — CHANGES_REQUESTED** [review]

File: apps/frontend/src/pages/profile/libs/components/update-avatar-form/update-avatar-form.tsx [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+type Properties = {
+	onSubmit: (data: FormData) => void;
+	user: UserDto;
+};
+
+const UpdateAvatarForm: React.FC<Properties> = ({
+	onSubmit,
+	user,
+}: Properties) => {
+	const fileInputReference = useRef<HTMLInputElement>(null);
```
Comment (fshabanov) [inline apps/frontend/src/pages/profile/libs/components/update-avatar-form/update-avatar-form.tsx:19]:
```suggestion
	const fileInputReference = useRef<HTMLInputElement | null>(null);
```

**Review by fshabanov — APPROVED** [review]

**Review by fshabanov — APPROVED** [review]

**Pull request merged**